### DISCUSSION TITLE
If a reboot is needed after every Store operation, should the firmware automatically reboot?

### DISCUSSION THREAD

**Discussion by Ergonomicmike** [mriscoc/Ender3V2S1#796] [Ideas]
The [Calibration Guide](https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides) repeatedly says - as a last step in most Sections - to "Store your settings and Control/Reboot printer."

I admit that I haven't been Rebooting. And the new Settings seem to persist.

Still, if a Reboot is really necessary, then perhaps the firmware should pop up a window ("Rebooting Now") and reboot itself?

**Comment by TheWebMachine** [comment]
I also have never rebooted after a Store operation or even when loading back from a config file (which does a Store at the end) and have never had a problem. 

This is either outdated guide info due to old firmware issues or perhaps was noted because we previously found that you had to set larger machine physical limits *before* it would let you set larger bed sizes (Plus models, etc). It may have been previously thought reboot was necessary to expand the bed size after physical limit adjustment.

Either way, a reboot doesn't seem necessary now (at least on F4 SOCs).

**Comment by mriscoc** [comment]
Marlin core assumes that the physical settings are invariants, that is, defined in code and not user modificable, so to ensure that the core takes into account all possible not standard changes I suggest, at the end of the calibration, do a reboot.

No need to reboot after simpler changes.

**Reply by TheWebMachine** [reply]
Actually, I do remember needing to reboot between setting machine physical limits and bed size/mesh inset when I *manually* altered the settings in a new firmware flash.

However, since moving to using a config.gco to reload my firmware settings after each flash, I haven't needed to reboot after loading the config.gco. Something about setting via gcode appears to make it apply everything without requiring the reboot. I've been able to flash, load config.gco, and jump right into a print no problem. Changing via menus manually still requires a reboot after physical limits before it will let you set the bed size and mesh inset...this is especially true if you need to set a larger bed size than default.

I wonder what causes the Marlin core to reload the values if loaded from a config file but not if changed in menus and Stored that way.

**Reply by csanii123** [reply]
I am looking to get my setup as much as I can in the config file to do the same thing. Could you post a copy of your config.gco so I can see how you did it ? I managed to get a few of my settings setup but I suspect I have others I should set but just not sure if or how to do it.

**Reply by TheWebMachine** [reply]
Sure! Do note the following:
* I have the Ender-3 S1 Plus, which is the model with the 300x300x300cm build volume and the F4 board. My values for C100-102 reflect manual measures of all min/max dimensions of the Plus model
* I am using the CR-Touch "zero offset" mount mod to move my probe to closer alignment with the print head: #614 https://www.printables.com/model/257460-ender-3-s1-zero-offset-cr-touch-mount In the case of the Plus model, the mod doesn't position to exactly 0 offset on the Y, which is reflected in my config below, but it definitely fixed my ABL/UBL/first layer issues. Assuming you are using all stock hotend, you will need to change those values accordingly (`M851`)
* I keep my brightness VERY low on the screen (`M256 B5`) because I use octoprint, so you may want to increase this to something like `B25` or higher so you can see the screen better
* I am relatively sure that **_THE ORDER OF OPTIONS MATTER!_** I don't think Marlin will accept a larger value for things like `C102` if it exceeds the current Min/Max. So you should always run `C100` and `C101` **first, before** `C102` and `C29` to ensure those latter options are within bounds
* I found that my print dimensional accuracy was a little off. So I have added `M92` to slightly adjust the X and Y steppers from the default 80 steps/mm to 79.6 steps/mm. You can confirm your stepper steps/mm are producing accurate prints by printing a cube of known size, measuring with calipers, and doing some basic math to determine the correct steps/mm...plenty of guides out there on exactly how to do this 
* I _intentionally_ include a 1 second delay at the end before storing. I was noticing that if I immediately issued `M500` after the last parameter change, there was a 50/50 shot the printer would randomly crash/reboot without saving anything. As soon as I added the `G4 S1` delay, this problem vanished
* The M117 messages show on the bottom of the LCD to let you know the status of the config apply; totally optional
* I have recently started adding options like `C412 M1` to _reinforce_ existing defaults...this is for my sanity in case a default changes in future versions for any reason

Anyway, here it is:
````
; Professional Firmware Configuration File
;=====================================================
C10                                  ; Mark as a configuration file
M117 Applying configuration
;-----------------------------------------------------
M256 B5                              ; Set Brightness 
M413 S0                              ; Disable Powerloss recovery (my FW doesn't have it and I have an UPS)
M603 L50.00 U50.00                   ; Configure Filament Change Load and Unload lengths in Filament Management menu
C851 S960 M0                         ; Probe feedrate and disable multiple probing
M851 X-45.4 Y-1.8 Z-1.75             ; Probe offsets
C100 X-8 Y-3                         ; Set XY Min position
C101 X323 Y310 Z310                  ; Set XYZ Max position
C102 X300 Y300                       ; Set a bed size
C29 L22.4 R277.6 F22.4 B277.6 N9 T0  ; Set Mesh insets, size of 9x9 and bed at 0°C
C125 X0 Y310 Z50                     ; Set Park Position
C412 M1                              ; Filament sensor active mode: HIGH
M92 X79.6 Y79.6                      ; Set Steps/mm
;-----------------------------------------------------
G4 S1                                ; Wait a second
M500			             ; save to EEPROM
M117 Configuration Applied
````

**Reply by neshami** [reply]
Your build volume is 27000 cm3.
E3S1 and pro have 13068 cm3

**Reply by TheWebMachine** [reply]
> Your build volume is 27000 cm3. E3S1 and pro have 13068 cm3

While I think everyone understood what I meant, I've corrected my comment to reflect what *I intended.*

**Comment by Ergonomicmike** [comment]
Okay. So in other words, for major changes in settings (like the size of the build plate), a reboot is necessary so that the firmware can "think" about it and recalculate any lower "children" variables/settings that are dependent on build plate size.
